test/prompt/invocation_layer/test_hugging_face.py

```diff
@@ -0,0 +1,424 @@
+from unittest.mock import MagicMock, patch, Mock
+
+import pytest
+from torch import device
+from transformers import AutoModelForSeq2SeqLM, AutoTokenizer, BloomForCausalLM, StoppingCriteriaList, GenerationConfig
+
+from haystack.nodes.prompt.invocation_layer import HFLocalInvocationLayer
+from haystack.nodes.prompt.invocation_layer.handlers import HFTokenStreamingHandler, DefaultTokenStreamingHandler
+from haystack.nodes.prompt.invocation_layer.hugging_face import StopWordsCriteria
+
+
+def create_mock_pipeline(model_name_or_path=None, max_length=100):
+    # mock transformers pipeline
+    return Mock(
+        **{"model_name_or_path": model_name_or_path},
+        return_value=Mock(**{"model_name_or_path": model_name_or_path, "tokenizer.model_max_length": max_length}),
+    )
+
+
+@pytest.mark.unit
+def test_constructor_with_invalid_task_name():
+    """
+    Test HFLocalInvocationLayer init with invalid task_name
+    """
+    # Create mocked transformers pipeline with a 'tokenizer' attribute (needed in HuggingFaceLocalInvocationLayer)
+    mock_pipeline = create_mock_pipeline()
+    mock_get_task = Mock(return_value="text2text-generation")
+
+    with patch("haystack.nodes.prompt.invocation_layer.hugging_face.get_task", mock_get_task):
+        with patch("haystack.nodes.prompt.invocation_layer.hugging_face.pipeline", mock_pipeline):
+            with pytest.raises(ValueError, match="Task name custom-text2text-generation is not supported"):
+                layer = HFLocalInvocationLayer("google/flan-t5-base", task_name="custom-text2text-generation")
```

Comment:
If you create the aforementioned fixture, tests can look like this:


```suggestion
def test_constructor_with_invalid_task_name(mock_pipeline):
    """
    Test HFLocalInvocationLayer init with invalid task_name
    """
    # Create mocked transformers pipeline with a 'tokenizer' attribute (needed in HuggingFaceLocalInvocationLayer)
    mock_get_task = Mock(return_value="text2text-generation")

    with patch("haystack.nodes.prompt.invocation_layer.hugging_face.get_task", mock_get_task):
        with pytest.raises(ValueError, match="Task name custom-text2text-generation is not supported"):
            layer = HFLocalInvocationLayer("google/flan-t5-base", task_name="custom-text2text-generation")
```